components/raftstore/src/store/fsm/peer.rs

```diff
@@ -481,6 +482,19 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         self.register_split_region_check_tick();
     }
 
+    fn on_capture_change(&mut self, cmd: ChangeCmd, cb: Callback<RocksEngine>) {
+        if !self.fsm.peer.is_leader() {
+            cb.invoke_with_response(new_error(Error::NotLeader(
+                self.region_id(),
+                self.fsm.peer.get_peer_from_cache(self.fsm.peer.leader_id()),
+            )));
+            return;
+        }
+        self.ctx
+            .apply_router
+            .schedule_task(self.region_id(), ApplyTask::Change { cmd, cb })
```

Comment:
Is it possible that the leader changes after this but before the cmd is executed?

Reply:
Yes, but role change events will be caught by the cdc observer.